File: lib/plans.ts
import type { Plan } from "@/lib/types";

export const planDetails: Record<Exclude<Plan, "FREE">, {
  name: string;
  price: string;
  listings: string;
  shops: string;
  planId: string;
  credits: string;
}> = {
  BASIC: {
    name: "Basic",
    price: "$19",
    listings: "20 listings/month",
    shops: "1 shop",
    planId: "basic",
    credits: "150 credits/month"
  },
  PRO: {
    name: "Pro",
    price: "$49",
    listings: "100 listings/month",
    shops: "3 shops",
    planId: "pro",
    credits: "600 credits/month"
  },
  AGENCY: {
    name: "Agency",
    price: "$99",
    listings: "Unlimited listings",
    shops: "10 shops",
    planId: "agency",
    credits: "2000 credits/month"
  }
};

export const planCredits: Record<Plan, string> = {
  FREE: "20 signup credits",
  BASIC: "150 credits/month",
  PRO: "600 credits/month",
  AGENCY: "2000 credits/month"
};

export function formatLimit(limit: number | null) {
  return limit === null ? "Unlimited" : limit.toString();
}

File: backend/app/core/plans.py
from backend.app.models.user import Plan

PLAN_LIMITS = {
    Plan.FREE: {"listings": 0, "shops": 1, "label": "Free"},
    Plan.BASIC: {"listings": 20, "shops": 1, "label": "Basic"},
    Plan.PRO: {"listings": 100, "shops": 3, "label": "Pro"},
    Plan.AGENCY: {"listings": None, "shops": 10, "label": "Agency"}
}


def normalize_plan(plan_id: str) -> Plan:
    normalized = plan_id.upper()
    if normalized not in Plan.__members__:
        raise ValueError("Unknown plan")
    return Plan[normalized]
